master/buildbot/process/log.py

```diff
@@ -176,12 +174,18 @@ def wholeLines(lines):
             return lbf
 
     def addStdout(self, text):
+        if not isinstance(text, str):
```

Comment:
at some point, I had in mind to do opportunistic decoding, so that we can have a fast path which send log data directly from worker to db when no logobserver is setup. I think that would help greatly in performance.

I think it is easier to have the decoding asap, though

Reply:
I haven't considered this, thanks for explaining.